server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -376,7 +392,7 @@ public static TransportVersion fromString(String str) {
      * </ul>
      */
     public boolean isPatchFrom(TransportVersion version) {
```

Comment:
It looks like there are only a handful of remaining uses, perhaps these versions could be manually migrated (building a chained TransportVersion in memory) so that they can use supports and this could be made private?